lib/topic_query/private_message_lists.rb

```diff
@@ -143,16 +143,20 @@ def private_messages_for(user, type)
       elsif type == :user
         result = result.where("topics.id IN (SELECT topic_id FROM topic_allowed_users WHERE user_id = #{user.id.to_i})")
       elsif type == :all
-        result = result.where("topics.id IN (
-              SELECT topic_id
-              FROM topic_allowed_users
-              WHERE user_id = #{user.id.to_i}
-              UNION ALL
-              SELECT topic_id FROM topic_allowed_groups
-              WHERE group_id IN (
-                SELECT group_id FROM group_users WHERE user_id = #{user.id.to_i}
-              )
-      )")
+        group_ids = group_with_messages_ids(user)
```

Comment:
Minor, but should you have the same "Executing an extra query...." comment here to stop people changing this? Or is it only the `filter_archived` query that suffers from the performance penalty?

Reply:
Only the `filtered_archived` query that will call this again which is why I extracted it into a method to memoize the values.

Reply:
Good point let me include that.